modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java

```diff
@@ -489,110 +428,175 @@ public static void send(
+                        callback.setTimeOutAction(endpoint.getTimeoutAction());
+                        callback.setTimeoutType(endpoint.getEndpointTimeoutType());
+                        if (log.isDebugEnabled()) {
+                            log.debug("Setting Timeout for endpoint : " +
+                                    getEndpointLogMessage(synapseOutMessageContext, axisOutMsgCtx) +
+                                    " to static timeout value : " + endpointTimeout);
+                        }
+                    } else {
+                        long endpointTimeout = endpoint.evaluateDynamicEndpointTimeout(synapseOutMessageContext);
+                        callback.setTimeout(endpointTimeout);
+                        callback.setTimeOutAction(endpoint.getTimeoutAction());
+                        callback.setTimeoutType(endpoint.getEndpointTimeoutType());
+                        if (log.isDebugEnabled()) {
+                            log.debug("Setting Timeout for endpoint : " +
+                                    getEndpointLogMessage(synapseOutMessageContext, axisOutMsgCtx) +
+                                    " to dynamic timeout value : " + endpointTimeout);
+                        }
+                    }
+                } else {
+                    long globalTimeout = synapseOutMessageContext.getEnvironment().getGlobalTimeout();
+                    callback.setTimeout(globalTimeout);
+                    callback.setTimeoutType(SynapseConstants.ENDPOINT_TIMEOUT_TYPE.GLOBAL_TIMEOUT);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Setting timeout for implicit endpoint : " +
+                                getEndpointLogMessage(synapseOutMessageContext, axisOutMsgCtx) +
+                                " to global timeout value of " + globalTimeout);
+                    }
+                }
+
+            }
+            mepClient.setCallback(callback);
 //
 //        if (Utils.isClientThreadNonBlockingPropertySet(axisOutMsgCtx)) {
 //            SynapseCallbackReceiver synapseCallbackReceiver = (SynapseCallbackReceiver) axisOutMsgCtx.getAxisOperation().getMessageReceiver();
 //            synapseCallbackReceiver.addCallback(axisOutMsgCtx.getMessageID(), new FaultCallback(axisOutMsgCtx, synapseOutMessageContext));
 //        }
 
-        // this is a temporary fix for converting messages from HTTP 1.1 chunking to HTTP 1.0.
-        // Without this HTTP transport can block & become unresponsive because we are streaming
-        // HTTP 1.1 messages and HTTP 1.0 require the whole message to caculate the content length
-        if (originalInMsgCtx.isPropertyTrue(NhttpConstants.FORCE_HTTP_1_0)) {
-            synapseOutMessageContext.getEnvelope().toString();
-        }
+            // this is a temporary fix for converting messages from HTTP 1.1 chunking to HTTP 1.0.
+            // Without this HTTP transport can block & become unresponsive because we are streaming
```

Comment:
Please don't do formatting in the existing code and include those in a PR of a feature. That makes the PR unreadable.